hydromt/models/v1/grid.py

```diff
@@ -0,0 +1,749 @@
+                If this is used, variables must be specified.
+
+        Parameters
+        ----------
+        vector_fn : str, Path, gpd.GeoDataFrame
+            Data catalog key, path to vector file or a vector geopandas object.
+        variables : List, str, optional
+            List of variables to add to grid from vector_fn. Required if
+            rasterize_method is "value", by default None.
+        nodata : List, int, float, optional
+            No data value to use for rasterization, by default -1. If a list is
+            provided, it should have the same length has variables.
+        rasterize_method : str, optional
+            Method to rasterize the vector data. Either {"value", "fraction", "area"}.
+            If "value", the value from the variables columns in vector_fn are used
+            directly in the raster. If "fraction", the fraction of the grid
+            cell covered by the vector file is returned. If "area", the area of the
+            grid cell covered by the vector file is returned.
+        mask_name: str, optional
+            Name of mask in self.grid to use for masking raster_fn. By default 'mask'.
+            Use None to disable masking.
+        rename: dict, optional
+            Dictionary to rename variable names in variables before adding to grid
+            {'name_in_variables': 'name_in_grid'}. To rename with method fraction or
+            area use {'vector_fn': 'name_in_grid'}. By default empty.
+        all_touched : bool, optional
+            If True (default), all pixels touched by geometries will be burned in.
+            If false, only pixels whose center is within the polygon or that are
+            selected by Bresenham's line algorithm will be burned in.
+
+        Returns
+        -------
+        list
+            Names of added model grid layers
+        """  # noqa: E501
+        rename = rename or dict()
+        self.logger.info(f"Preparing grid data from vector '{vector_fn}'.")
+        gdf = data_catalog.get_geodataframe(
+            vector_fn, geom=self.region, dst_crs=self.crs
+        )
+        if gdf.empty:
+            self.logger.warning(
+                f"No shapes of {vector_fn} found within region,"
+                " skipping setup_grid_from_vector."
+            )
+            return
+        # Data resampling
+        if vector_fn in rename.keys():
+            # In case of choosing a new name with area or fraction method pass
+            # the name directly
+            rename = rename[vector_fn]
+        ds = workflows.grid.grid_from_geodataframe(
+            grid_like=self._data,
+            gdf=gdf,
+            variables=variables,
+            nodata=nodata,
+            rasterize_method=rasterize_method,
+            mask_name=mask_name,
+            rename=rename,
+            all_touched=all_touched,
+        )
+        # Add to grid
+        self.set(ds)
+
+        return list(ds.data_vars.keys())
+
+    def setup_grid(
```

Comment:
This is actually your create function above which for now you have RaiseNotImplemented :)